source/gui/addonGui.py

```diff
@@ -297,29 +548,191 @@ def onEnableDisable(self, evt):
+			"Please contact the add-on author for further assistance."
+		).format("%s.%s.%s" % NVDAVersion)
+		AddonSelectionIntroLabel=wx.StaticText(self, label=introText)
+		AddonSelectionIntroLabel.Wrap(self.scaleSize(maxControlWidth))
+		sHelper.addItem(AddonSelectionIntroLabel)
+		# Translators: the label for the addons list in the addon selection dialog.
+		entriesLabel=_("Untested add-ons")
+		self.addonsList = sHelper.addLabeledControl(
+			entriesLabel,
+			nvdaControls.AutoWidthColumnCheckListCtrl,
+			style=wx.LC_REPORT|wx.LC_SINGLE_SEL,
+			size=self.scaleSize((maxControlWidth, 350))
+		)
+
+		# Translators: The label for a column in add-ons list used to check / uncheck whether the add-on is manually set  compatible
+		self.addonsList.InsertColumn(0, _("Available"), width=self.scaleSize(70))
+		# Translators: The label for a column in add-ons list used to identify add-on package name (example: package is OCR).
+		self.addonsList.InsertColumn(1, _("Package"), width=self.scaleSize(150))
+		# Translators: The label for a column in add-ons list used to identify add-on's running status (example: status is running).
+		self.addonsList.InsertColumn(2, _("Version"), width=self.scaleSize(150))
+		# Translators: The label for a column in add-ons list used to identify the last version of NVDA
+		# this add-on has been tested with.
+		self.addonsList.InsertColumn(3, _("Last tested NVDA version"), width=self.scaleSize(180))
+
+		buttonSizer = guiHelper.ButtonHelper(wx.HORIZONTAL)
+		# Translators: The continue button on a NVDA dialog. This button will accept any changes and dismiss the dialog.
+		continueID = wx.ID_OK
+		button = buttonSizer.addButton(self, label=_("Continue"), id=continueID)
+		sHelper.addDialogDismissButtons(buttonSizer)
+		mainSizer.Add(settingsSizer, border=20, flag=wx.ALL)
+		mainSizer.Fit(self)
+		self.SetSizer(mainSizer)
+
+		self.SetAffirmativeId(continueID)
+		self.SetEscapeId(continueID)
+		button.Bind(wx.EVT_BUTTON, self.onContinue)
+
+		self.refreshAddonsList()
+		self.addonsList.SetFocus()
+		self.CentreOnScreen()
+
+	def refreshAddonsList(self,activeIndex=0):
+		self.addonsList.DeleteAllItems()
+		self.curAddons=[]
+		for idx, addon in enumerate(self.unknownCompatibilityAddonsList):
+			self.addonsList.Append((
+				"",  # check box field
+				addon.manifest['summary'],
+				addon.version,
+				addon.manifest.get('lastTestedNVDAVersion') or
+				# Translators: Displayed for add-ons in the incompatible add-ons dialog
+				# if the last tested NVDA version is not specified.
+				_("not specified")
+			))
+			compatValue = AddonCompatibilityState.getAddonCompatibility(addon, CURRENT_NVDA_VERSION)
+			shouldCheck = compatValue == compatValues.MANUALLY_SET_COMPATIBLE
+			self.addonsList.CheckItem(idx, check=shouldCheck)
+		# select the given active addon or the first addon if not given
+		curAddonsLen=len(self.unknownCompatibilityAddonsList)
+		if curAddonsLen>0:
+			if activeIndex==-1:
+				activeIndex=curAddonsLen-1

```

Comment:
I do not understand this code. When would an index be -1 and why should we select the last add-on in the lit in this case?

Reply:
`activeIndex` is passed in as `-1` from https://github.com/nvaccess/nvda/blob/3d6ad08dfa23d58df50b5660c77b0f2e5af25560/source/gui/addonGui.py#L389 after an addon has been successfully installed. This results in the last addon in the list being selected. I'll add something to the comment for this.